src/pipx/venv_inspect.py

```diff
@@ -0,0 +1,268 @@
+                venv_inspect_info.bin_path / app for app in app_names
+            ]
+        # recursively search for more
+        dep_visited[dep_name] = True
+        app_paths_of_dependencies = _dfs_package_apps(
+            dep_dist, dep_req, venv_inspect_info, app_paths_of_dependencies, dep_visited
+        )
+    return app_paths_of_dependencies
+
+
+def _windows_extra_app_paths(app_paths: List[Path]) -> List[Path]:
+    # In Windows, editable package have additional files starting with the
+    #   same name that are required to be in the same dir to run the app
+    # Add "*-script.py", "*.exe.manifest" only to app_paths to make
+    #   execution work; do not add them to apps to ensure they are not listed
+    app_paths_output = app_paths.copy()
+    for app_path in app_paths:
+        win_app_path = app_path.parent / (app_path.stem + "-script.py")
+        if win_app_path.exists():
+            app_paths_output.append(win_app_path)
+        win_app_path = app_path.parent / (app_path.stem + ".exe.manifest")
+        if win_app_path.exists():
+            app_paths_output.append(win_app_path)
+    return app_paths_output
+
+
+def fetch_info_in_venv(venv_python_path) -> Tuple[List[str], Dict[str, str], str]:
+    command_str = textwrap.dedent(
+        """
+        import json
+        import os
+        import platform
+        import sys
+
+        impl_ver = sys.implementation.version
+        implementation_version = "{0.major}.{0.minor}.{0.micro}".format(impl_ver)
+        if impl_ver.releaselevel != "final":
+            implementation_version += impl_ver.releaselevel[0] + impl_ver.serial
+        print(
+            json.dumps(
+                {
+                    'sys_path':sys.path,
+                    'python_version':"{0.major}.{0.minor}.{0.micro}".format(sys.version_info),
+                    'environment': {
+                        "implementation_name": sys.implementation.name,
+                        "implementation_version": implementation_version,
+                        "os_name": os.name,
+                        "platform_machine": platform.machine(),
+                        "platform_release": platform.release(),
+                        "platform_system": platform.system(),
+                        "platform_version": platform.version(),
+                        "python_full_version": platform.python_version(),
+                        "platform_python_implementation": platform.python_implementation(),
+                        "python_version": ".".join(platform.python_version_tuple()[:2]),
+                        "sys_platform": sys.platform,
+                    }
+                }
+            )
+        )
+        """
+    )
+    venv_info = json.loads(
+        run_subprocess(
+            [venv_python_path, "-c", command_str],
+            capture_stderr=False,
+            log_cmd_str="<inline fetch_info_in_venv commands>",
+        ).stdout
+    )
+    return (
+        venv_info["sys_path"],
```

Comment:
Just realized that `sys.path` is returning the first entry as an empty string, which should probably be removed, since when we are fetching distributions we don't want to include the "current directory".